BrowserKit/Sources/Common/Extensions/UIButton.ConfigurationExtension.swift

```diff
@@ -0,0 +1,18 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+
+extension UIButton.Configuration {
```

Comment:
We are trying to [move away from using extensions](https://github.com/mozilla-mobile/firefox-ios/wiki/When-To-Use-Extensions) especially if they are only used a few times throughout the app. Could you please remove this one?

Reply:
This is the first PR of a series to come to migrate towards using `UIButton.Configuration` throughout the app so I thought it made sense to add an extension for it. I've counted at least 17 places where the font should be set using `titleTextAttributesTransformer`, do you think it is still worth it to duplicate those 5 lines of code in all those places? If so I will gladly remove it 🙂

Reply:
What I see is that currently there are 11 buttons that are just plain `LegacyResizableButton` with some custom styling (cmd+shift+f `: LegacyResizableButton =`), some use `ActionButton` and finally some are just plain `UIButton` but since they use deprecated insets we should also use `UIButton.Configuration` for them. 
However, maybe the solution is to replace all of those with `PrimaryRoundedButton`, `SecondaryRoundedButton` or `LinkButton` depending on which is relevent in each of those cases, you tell me. In that case I do agree that the extension is unnecessary 👍

Reply:
@lmarceau would be able to confirm that. She was creating tasks to replace the old buttons with the component library ones.

Reply:
I think this is fine as a quick incremental step just to silence the warnings. Moving these legacy buttons to use components would be a bigger task and might be a long time before we get around to it. Agreed tho that once everything is using the component buttons the extension should no longer be necessary.

Reply:
I would agree that a first step to silence the warnings without doing too much of a refactor might be a good idea. Perhaps I can add a comment to the extension stating that it is intended to be removed later on as we move forward to using button components?

Reply:
Done in https://github.com/mozilla-mobile/firefox-ios/pull/17616/commits/2ce44deedafa246679e7d56e7b2ef0f25db97662! 🙂